examples/static_set/device_reference_example.cu

```diff
@@ -34,8 +34,8 @@ __global__ void custom_cooperative_insert(SetRef raw_set, InputIterator keys, st
 
   constexpr auto cg_size = SetRef::cg_size;
 
-  // we haven't spcified any functions yet so we make a copy with the desired functionality
-  auto set = raw_set.template with_functions<cuco::experimental::insert>();
+  // we haven't spcified any operatorss yet so we make a copy with the desired functionality
+  auto set = raw_set.template with<cuco::experimental::insert>();
```

Comment:
Oh, gross. I didn't think about how this design would lead to needing a template disambiguator.

The usual solution to avoid this is to make the function a free function instead of a member function. That is why things like `std::get` are a free function and not a member of `tuple`.

I'm not sure if applies as well to our situation, but I really dislike that this will end up requiring most users to use a template disambiguator.

Reply:
Yeah, @sleeepyjack also mentioned this issue https://github.com/NVIDIA/cuCollections/pull/246#discussion_r1038905868

Reply:
Right. I'll switch to a free function.

Reply:
Oh, this is going to be trickier than I thought. We would need to partially specialize `cuco::with<Reference, Operators...>` on the reference type, which is not possible with function templates. Let's see if I can make it work with a functor-like thing.

Reply:
Also fixed typo in [63ed50f](https://github.com/NVIDIA/cuCollections/pull/246/commits/63ed50fbe1e4ad1d99fa42de86e087b4f3ebbaf0)